### PR TITLE
Define ANDROID_ABI in Cobalt

### PR DESCRIPTION
**Pull request by haozheng-cobalt**
b/385380562

### CODE REVIEW

File: cobalt/user_agent/BUILD.gn [hunk 1]
```diff
@@ -24,4 +24,12 @@ static_library("user_agent") {
     "//starboard/common",
     "//v8:v8_version",
   ]
+
+  if (is_android) {
+    import("//build/config/android/abi.gni")
+    defines = [
+      # Enable compile-time decisions based on the ABI
+      "ANDROID_ABI=$android_app_abi",
```
Comment (andrewsavage1) [inline cobalt/user_agent/BUILD.gn:32]: I also see this used in https://github.com/youtube/cobalt/blob/main/starboard/android/shared/system_get_property.cc#L39, is that file no longer used?

File: cobalt/user_agent/BUILD.gn [hunk 2]
```diff
@@ -24,4 +24,12 @@ static_library("user_agent") {
     "//starboard/common",
     "//v8:v8_version",
   ]
+
+  if (is_android) {
+    import("//build/config/android/abi.gni")
```
Comment (andrewsavage1) [inline cobalt/user_agent/BUILD.gn:29]: Import this at the top of the file

File: cobalt/user_agent/BUILD.gn (hunk 1 above)
Comment (haozheng-cobalt) [reply cobalt/user_agent/BUILD.gn:35]: yes, that function is no longer used, this file can be cleaned up or fully migrated in the future.

File: cobalt/user_agent/BUILD.gn [hunk 3]
```diff
@@ -24,4 +28,11 @@ static_library("user_agent") {
     "//starboard/common",
     "//v8:v8_version",
   ]
+
+  if (is_android) {
+    defines = [
+      # Enable compile-time decisions based on the ABI
+      "ANDROID_ABI=$android_app_abi",
```
Comment (andrewsavage1) [inline cobalt/user_agent/BUILD.gn:35]: Double check that this macro is being set to what you expect (i.e. aligns with what the platform_configuration files were setting the android_abi to)

**Review by andrewsavage1 — APPROVED** [review]
LGTM as long as you've tested the values are correct for our configurations

File: cobalt/user_agent/BUILD.gn (hunk 3 above)
Comment (haozheng-cobalt) [reply cobalt/user_agent/BUILD.gn:35]: Yes, I've already verified the field locally, and fortunately android_app_abi align with what the platform_configuration file would set.

**Pull request merged**